### ISSUE TITLE
Add inline docs in C headers

### ISSUE THREAD

**Issue by sffc** [rust-diplomat/diplomat#101]
From @dminor in https://github.com/unicode-org/icu4x/pull/1269#discussion_r747045186:

> Is it intentional for the C++ headers to include comments, but not the C headers? It makes sense that the copies of the C headers in ffi/diplomat/cpp are just an implementation detail of the C++ headers, but the same argument doesn't seem to hold for ffi/diplomat/c.

Note: @gregtatum added C++ docs generation in https://github.com/rust-diplomat/diplomat/pull/85.  The primary documentation is intended to be the `*.rst` files, but perhaps we should put the docs into the `*.h` files as well.

**Comment by gregtatum** [comment]
Yeah, this would be really nice for in-editor hints. There was no particularly reason for not doing C headers, other than it was more work that just doing C++ headers.
